crates/types/src/config/common.rs

```diff
@@ -853,106 +853,22 @@ pub struct CommonOptionsShadow {
     pub default_replication: ReplicationProperty,
 
     metadata_client: MetadataClientOptions,
-    // todo drop in version 1.3
-    #[serde_as(as = "Option<MetadataClientOptionsWithConfigurationDefaults>")]
-    metadata_store_client: Option<MetadataClientOptions>,
-    #[serde_as(as = "Option<serde_with::DisplayFromStr>")]
-    // todo drop in version 1.3
-    metadata_store_connect_timeout: Option<humantime::Duration>,
-    #[serde_as(as = "Option<serde_with::DisplayFromStr>")]
-    // todo drop in version 1.3
-    metadata_store_keep_alive_interval: Option<humantime::Duration>,
-    #[serde_as(as = "Option<serde_with::DisplayFromStr>")]
-    // todo drop in version 1.3
-    metadata_store_keep_alive_timeout: Option<humantime::Duration>,
-    // todo drop in version 1.3
-    metadata_store_backoff_policy: Option<RetryPolicy>,
 
     auto_provision: bool,
-    // todo drop in version 1.3
-    allow_bootstrap: Option<bool>,
 
     default_num_partitions: u16,
-    // todo drop in version 1.3
-    bootstrap_num_partitions: Option<u16>,
 }
 
 impl From<CommonOptionsShadow> for CommonOptions {
```

Comment:
Do we still need the `CommonOptionsShadow`?

Reply:
yes, it's unused; i'd prefer keep it for use when parameter conversion is needed later.